pkg/controller/seed-controller-manager/initial-application-installation-controller/controller.go

```diff
@@ -224,3 +240,16 @@ func (r *Reconciler) removeAnnotation(ctx context.Context, cluster *kubermaticv1
 	delete(cluster.Annotations, kubermaticv1.InitialApplicationInstallationsRequestAnnotation)
 	return r.Patch(ctx, cluster, ctrlruntimeclient.MergeFrom(oldCluster))
 }
+
+func getCNIApplicationInstallation(ctx context.Context, userClusterClient ctrlruntimeclient.Client, cniType kubermaticv1.CNIPluginType) (*appskubermaticv1.ApplicationInstallation, error) {
+	app := &appskubermaticv1.ApplicationInstallation{}
+	switch cniType {
+	case kubermaticv1.CNIPluginTypeCilium:
+		name := kubermaticv1.CNIPluginTypeCilium.String()
+		if err := userClusterClient.Get(ctx, types.NamespacedName{Namespace: metav1.NamespaceSystem, Name: name}, app); err != nil {
+			return nil, fmt.Errorf("failed to get Cilium ApplicationInstallation in user cluster: %w", err)
+		}
+	}
```

Comment:
Sorry, one more addition here: Can you please add a `default` case that returns an "unsupported" error? Else we return the `nil` pointer `app` variable in the end, don't think we should do that.

Reply:
Hmm, sure we can do that, we are checking earlier if cni.IsManagedByAppInfra.
Also we can just return this unsupported error after switch statement

Reply:
I agree that it shouldn't happen, but IMO it's a good idea to do it anyway to exhaust all possible inputs. I don't have strong opinions whether it's in a `default` block or after the switch statement.